.changeset/pre.json

```diff
@@ -0,0 +1,42 @@
+{
+  "mode": "pre",
+  "tag": "preview",
+  "initialVersions": {
+    "@solana/accounts": "2.0.0-preview.1",
+    "@solana/addresses": "2.0.0-preview.1",
+    "@solana/assertions": "2.0.0-preview.1",
+    "@solana/codecs-core": "2.0.0-preview.1",
+    "@solana/codecs-data-structures": "2.0.0-preview.1",
+    "@solana/codecs-numbers": "2.0.0-preview.1",
+    "@solana/codecs-strings": "2.0.0-preview.1",
+    "@solana/codecs": "2.0.0-preview.1",
+    "@solana/compat": "2.0.0-preview.1",
+    "@solana/errors": "2.0.0-preview.1",
+    "@solana/functional": "2.0.0-preview.1",
+    "@solana/instructions": "2.0.0-preview.1",
+    "@solana/keys": "2.0.0-preview.1",
+    "@solana/options": "2.0.0-preview.1",
+    "@solana/programs": "2.0.0-preview.1",
+    "@solana/rpc-api": "2.0.0-preview.1",
+    "@solana/rpc-graphql": "2.0.0-preview.1",
+    "@solana/rpc-parsed-types": "2.0.0-preview.1",
+    "@solana/rpc-spec-types": "2.0.0-preview.1",
+    "@solana/rpc-spec": "2.0.0-preview.1",
+    "@solana/rpc-subscriptions-api": "2.0.0-preview.1",
+    "@solana/rpc-subscriptions-spec": "2.0.0-preview.1",
+    "@solana/rpc-subscriptions-transport-websocket": "2.0.0-preview.1",
+    "@solana/rpc-subscriptions": "2.0.0-preview.1",
+    "@solana/rpc-transformers": "2.0.0-preview.1",
+    "@solana/rpc-transport-http": "2.0.0-preview.1",
+    "@solana/rpc-types": "2.0.0-preview.1",
+    "@solana/rpc": "2.0.0-preview.1",
+    "@solana/signers": "2.0.0-preview.1",
+    "@solana/transaction-confirmation": "2.0.0-preview.1",
+    "@solana/transactions": "2.0.0-preview.1",
+    "@solana/web3.js": "0.0.0-development",
+    "@solana/web3.js-experimental": "2.0.0-preview.1",
+    "@solana/web3.js-legacy-sham": "1.0.0-preview.1",
+    "@solana/webcrypto-ed25519-polyfill": "2.0.0-preview.1"
```

Comment:
All of these versions have been updated to `-preview.1`. The first publish we do will take it to `-preview.2` and we'll call that TP2. I _think_ what we'll do after TP2 is released is _not_ do any releases until TP3, and if we need a release then we'll do a [changeset snapshot](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md).

Reply:
Or maybe we could install this: https://github.com/seek-oss/changesets-snapshot

Reply:
Is it possible to have a pre-release number with two numbers? like `2.0.0-preview.2.42`.

Reply:
I agree on continuous delivery. Surely there is a way to do what Loris is suggesting?

Reply:
Yep! That's snapshots. It'll make versions like `0.0.0-preview-20240307210610`. I'll install that in a follow up PR.

Reply:
For your amusement: https://github.com/solana-labs/solana-web3.js/commit/069e0006f493b83e842579fada1c983874f245b7#r139530996